internal/expr/expr_test.go

```diff
@@ -297,3 +314,144 @@ func FuzzParser(f *testing.F) {
 		parser.Parse(s)
 	})
 }
+
+func (s *ExprSuite) TestValidPrepare(c *C) {
+	testList := []struct {
+		input            string
+		prepareArgs      []any
+		expectedPrepared string
+	}{{
+		"SELECT street FROM t WHERE x = $Address.street",
+		[]any{Address{}},
+		"SELECT street FROM t WHERE x = ?",
+	}, {
+		"SELECT p FROM t WHERE x = $Person.id",
+		[]any{Person{}},
+		"SELECT p FROM t WHERE x = ?",
+	}}
+	for _, test := range testList {
+		parser := expr.NewParser()
+		parsedExpr, _ := parser.Parse(test.input)
+		preparedExpr, err := parsedExpr.Prepare(test.prepareArgs...)
+
+		if err != nil {
+			c.Fatal(err)
+		}
+		c.Assert(preparedExpr.SQL, Equals, test.expectedPrepared)
+	}
+}
+
+func (s *ExprSuite) TestPrepareMismatchedStructName(c *C) {
+	testList := []struct {
+		sql     string
+		structs []any
+	}{{
+		sql:     "SELECT street FROM t WHERE x = $Address.street",
+		structs: []any{Person{ID: 1}},
+	}, {
+		sql:     "SELECT street AS &Address.street FROM t",
+		structs: []any{},
+	}, {
+		sql:     "SELECT street AS &Address.id FROM t",
+		structs: []any{Person{ID: 1}},
+	}}
+
+	for i, test := range testList {
+		parser := expr.NewParser()
+		parsedExpr, err := parser.Parse(test.sql)
```

Comment:
Actually it'll probably be better if I put this in
```golang
if err != nil {
	c.Fatal(err)
}
```
Since we get a null pointer expression for some reason that did happen to be an error.

Reply:
sounds good!